docker-compose-dev.yaml

```diff
@@ -0,0 +1,120 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+version: '2.1'
+
+services:
+  processor:
+    build:
+      context: .
+      dockerfile: ./processor/Dockerfile-dev
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    image: marketplace-tp-dev
+    volumes:
+      - '.:/project/sawtooth-marketplace'
+    command: echo "No transaction processor command set"
+
+  server:
+    build:
+      context: .
+      dockerfile: ./server/Dockerfile-dev
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    image: marketplace-server-dev
+    volumes:
+      - .:/project/sawtooth-marketplace
+    ports:
```

Comment:
All ports exposed to the host, and all docker container names generated should be distinct and separate from any other docker compose file we use, so it can be used in parallel.  For example, it should be easy to run supply chain and mktplace at the same time without any user modification to the docker compose file.